src/useAsync.ts

```diff
@@ -18,14 +18,18 @@ export type AsyncState<T> =
       value: T;
     };
 
-const useAsync = <T>(fn: () => Promise<T>, deps: DependencyList = []) => {
-  const [state, callback] = useAsyncFn(fn, deps);
+export default function useAsync<
+  Result = any,
```

Comment:
this level of splitting up of the types allows everything to be type-aware, even the args for the callback.
previously, everything was any all the things